### PR TITLE
refactor(ecam): Convert COND page to React and add new SimVars

### PR DESCRIPTION
**Pull request by RichardPilbery**
Fixes #[issue_no]

## Summary of Changes

Refactor of existing COND page to use React and useSimVars. Changes should be invisible and not affect performance.

## Screenshots (if necessary)

cond

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Spawn in
2. Adjust cockpit air conditioning knobs and ensure COND page gauges and values match
3. Check when HOT AIR button is turned off, this change is reflected by the valve on the COND page.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: src/instruments/src/SD/Pages/Cond/Cond.tsx [hunk 1]
```diff
@@ -17,71 +17,60 @@
...
-    });
+    const [cockpitSelectedTemp] = useSimVar('L:A320_Neo_AIRCOND_LVL_1', 'number', 1000);
+    const [cockpitTrimTemp] = useSimVar('L:A32NX_CKPT_TRIM_TEMP', 'celsius', 1000);
+    const [cockpitCabinTemp] = useSimVar('L:A32NX_CKPT_TEMP', 'celsius', 1000);
+
+    const [cockpitRotateTemp, setCockpitRotateTemp] = useState(0);
+    useEffect(() => {
+        const rotate : number = gaugeOffset + cockpitSelectedTemp;
+        setCockpitRotateTemp(rotate);
+    }, [cockpitSelectedTemp]);
```
Comment (beheh) [inline src/instruments/src/SD/Pages/Cond/Cond.tsx:37]: This is not great, but of course this will be replaced by a real COND system simulation in future.

**Review by beheh — APPROVED** [review]

File: src/instruments/src/SD/Pages/Cond/Cond.tsx (hunk 1 above)
Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Cond/Cond.tsx:37]: Agree. Open to alternatives if there are any.

Comment (beheh) [reply src/instruments/src/SD/Pages/Cond/Cond.tsx:37]: For now you could just drop the duplicate state and just calculate `gaugeOffset + cockpitSelectedTemp` on every render. In future we probably won't have useEffect here either, but a separate, second LVar.

**Comment by marcman86** [comment]
Spawned

**Comment by RichardPilbery** [comment]
This is a separate error that is hopefully going to be addressed by #3753 . Did the page function correctly?

**Comment by marcman86** [comment]
Just after spawn it is a black screen that shows "If you see this, react didn't load"

**Comment by RichardPilbery** [comment]
If you select the COND page, what happens?

**Comment by beheh** [comment]
I just added the latest commit from master here for page registration fix.

**Comment by RichardPilbery** [comment]
Thanks @beheh : sounds like the DOORS page is not loading properly. I'll have a look tonight.

**Comment by beheh** [comment]
@RichardPilbery Let me know once you think this is ready to test again.

**Comment by RichardPilbery** [comment]
> @RichardPilbery Let me know once you think this is ready to test again.

It's ready @beheh

**Comment by Benjozork** [comment]
@beheh have any objections to me sending this to QA ?

**Review by beheh — CHANGES_REQUESTED** [review]
Just one more nitpick, otherwise this is good to go.

File: src/instruments/ecam-page-template/template.js [hunk 2]
```diff
@@ -63,19 +62,23 @@ class A32NX_PAGE_NAME_Logic extends Airliners.EICASTemplateElement {
...
+            const code = document.getElementById('A32NX_PAGE_NAME_BUNDLED_LOGIC').innerHTML;
             const script = document.createElement('script');
             script.innerHTML = code;
             document.body.appendChild(script);
         }
     }
 
+    onEvent(_event) {
+        console.log(JSON.stringify(_event));
+    }
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/instruments/ecam-page-template/template.js:80]: Whoops !

**Review by beheh — APPROVED** [review]

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 13.3.2021
Version of the sim: 1.14.5.0
PR Tested: #3766
Tier of testing: 1
Changes to observe: COND page
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP
2.) Start up AC
3.) Tested DOOR and COND page functions

Issues: -

Overall Rating: Good

Conclusions of the testing: All functions work. 

Question: Where is HOT AIR provided from? Is is available by just external power? When it is off, what temperture is the cond. air? It shows 18°C. Without heating, should it be something more like outside temperature?

Hope this helps :-)

**Comment by RichardPilbery** [comment]
It's not modelled yet. Might be included when the packs are sorted perhaps?

**Comment by darkfury334** [comment]
Category: QA Trial
Name: DarkFury#4936
Date of testing: Current date 13/03/2021
Version of the sim: 1.14.5.0
PR Tested: #3766
Tier of testing: 1
Changes to observe: Observe changes in COND page 


Testing technique: Adjust cockpit air conditioning knobs and ensure COND page match

Steps you've made to test the Pull Request:
1. Spawned in Cold & Dark at EGCC
2. Powered aircraft and started AC
3. Adjusted air condition and check COND Page functions
4. Turned hot air off and check COND page see if valve changed.

Flight notes (When done a full flight):N/A

Media: 

Issues (If Any):


Overall Rating: Good
Conclusions of the testing: All working

**Comment by ChruutvoLuzi** [comment]
Category: QA Tester
Name: ChruutvoLuzi | SD-PB.ch#8902
Date of testing: 13.3.2021
Version of the sim: 1.14.5.0
PR Tested: #3766 
Tier of testing: 1
Changes to observe: React Cond Page


Testing technique: TryOut
Steps you've made to test the Pull Request:
1. Spawned in LSZH
2. Checked various states and positions from knobs and buttons and checked if they were synced with the ecam

Media:
Microsoft Flight Simulator Screenshot 2021 03 13 - 19 23 56 32
Microsoft Flight Simulator Screenshot 2021 03 13 - 19 24 05 03


Issues (If Any): None

Overall Rating: Good
Conclusions of the testing: Works as described, looks like it did before.

**Pull request merged**